handlers.py

```diff
@@ -239,3 +238,52 @@ def handle_login(server, body, source):
         message = "Password is incorrect"
         print(message)
         server.send(Error.json(Error.BAD_REQ, message), source)
+
+
+def handle_set_led(server, body, source):
+    """
+    Sends SET_LED message to the destination that corresponds with the source
+    address in the cache.
+    """
+    print('SET_LED')
+
+    if 'state' not in body:
+        print('Missing field: state')
+        server.send(Error.json(Error.BAD_REQ, 'missing field: state'), source)
+        return
+
+    state = body['state']
+    if state < 0 or state > 2:
+        msg = 'state must be an int in range [0,2]'
+        print(msg)
+        server.send(Error.json(Error.BAD_REQ, msg), source)
+        return
+
+    server.add_route(source, source)
```

Comment:
What is this line for? A LINK request should've been sent before this so `add_route` shouldn't need to be called.

Reply:
Oh lol I put those lines in to test the handlers. Yeah I'll take them out, whoops! Good thing we have code reviews.